runtime/src/drivers.rs

```diff
@@ -114,31 +105,110 @@ impl Drivers {
                 .map_err(|_| CaliptraError::RUNTIME_INITIALIZE_DPE_FAILED)?;
             let env = DpeEnv::<CptraDpeTypes> {
                 crypto,
-                platform: DpePlatform::new(locality, hashed_rt_pub_key, &mut cert_chain),
+                platform: DpePlatform::new(locality, hashed_rt_pub_key, &mut drivers.cert_chain),
             };
-            let dpe = Self::initialize_dpe(env, &mut pcr_bank, locality)?;
-            persistent_data.get_mut().dpe = dpe;
+            let dpe = Self::initialize_dpe(env, &mut drivers.pcr_bank, locality)?;
+            drivers.persistent_data.get_mut().dpe = dpe;
         }
 
+        if reset_reason != ResetReason::ColdReset {
```

Comment:
nit: I think it might be easier to follow what's happening here with a bit of a refactor. Roughly

```rust
match reset_reason {
  ResetReason::ColdReset => {
    Self::initialize_dpe()?;
  },
  ResetReason::UpdateReset => {
    Self::check_dpe_structure()?;
    Self::update_dpe_rt_journey()?;
  },
  ResetReason::WarmReset => {
    Self::check_dpe_structure()?;
    Self::check_dpe_rt_journey_unchanged()?;
  }
}
```

Reply:
Except I suppose for warm and update reset, instead of early return we want to call `disable_attestation` and keep booting.

Reply:
Done